From 24039401245c6ba3bfb991e690a8ad85480845a4 Mon Sep 17 00:00:00 2001
From: Steve Kowalik <steven@wedontsleep.org>
Date: Tue, 16 Jan 2024 15:43:17 +1100
Subject: [PATCH] Do not unilaterally import distutils in setup.py

Python 3.12 removed distutils completly, so also attempt to import
Extension from setuptools, and if that fails, to import it from
distutils.
---
 setup.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index 1d420b8..292ea8f 100755
--- a/setup.py
+++ b/setup.py
@@ -7,10 +7,9 @@
 # part of the icecast project, http://svn.xiph.org/icecast/trunk/shout-python)
 
 try:
-    from setuptools import setup
+    from setuptools import setup, Extension
 except ImportError:
-    from distutils.core import setup
-from distutils.core import Extension
+    from distutils.core import setup, Extension
 import os
 import sys
 
